firmware/src/main.cpp

```diff
@@ -9,6 +9,16 @@
 #include <reg_types.h>
 #include "core1_sensor.h"
 
+#include "hardware/pll.h"
+#include "hardware/clocks.h"
+#include "hardware/structs/pll.h"
+#include "hardware/structs/clocks.h"
+
+#define SYS_CLOCK_SPEED_MHZ 12
```

Comment:
For testing, I suggest we connect the environment sensor to a white rabbit (time synchronizer) and confirm that the two devices can sync relative to each other. (Just spot-checking that the timestamps adjust to take on the white-rabbit time should be sufficient.)

Changing the system clock's speed should not affect any of the synchronization code, but this fix is a dealbreaker if it does since synchronization is one of the reasons for making this device harp-compliant in the first place.